src/timeseriesinsights/azext_timeseriesinsights/_params.py

```diff
@@ -0,0 +1,88 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    name_type,
+    tags_type,
+    get_enum_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+
+from knack.arguments import CLIArgumentType
+from ._validators import iso_8601_timespace
+
+environment_name_type = CLIArgumentType(id_part='name', help='The name of the Time Series Insights environment associated with the specified resource group.')
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('timeseriesinsights operation list') as c:
+        pass
+
+    with self.argument_context('timeseriesinsights') as c:
+        c.argument('location', get_location_type(self.cli_ctx), validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type)
+
+    # region environment
+    with self.argument_context('timeseriesinsights environment') as c:
+        from .vendored_sdks.timeseriesinsights.models import SkuName
+        c.argument('environment_name', arg_type=name_type, id_part='name', help='The name of the Time Series Insights environment associated with the specified resource group.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.argument('sku_name', arg_group="SKU", arg_type=get_enum_type(SkuName), help='The sku determines the type of environment, either standard (S1 or S2) or long-term (L1). For standard environments the sku determines the capacity of the environment, the ingress rate, and the billing rate.')
+        c.argument('sku_capacity', type=int, arg_group="SKU", help='The capacity of the sku. For standard environments, this value can be changed to support scale out of environments after they have been created.')
+
+    with self.argument_context('timeseriesinsights environment standard') as c:
+        from .vendored_sdks.timeseriesinsights.models import StorageLimitExceededBehavior
+        c.argument('storage_limit_exceeded_behavior', arg_type=get_enum_type(StorageLimitExceededBehavior))
+        c.argument('data_retention_time', type=iso_8601_timespace, help='The minimum number of days the environment\'s events will be available for query.')
+        c.argument('partition_key_properties', nargs='+')
+
+    with self.argument_context('timeseriesinsights environment longterm') as c:
+        c.argument('storage_account_name', arg_group="Storage Configuration", help='The name of the storage account that will hold the environment\'s long term data.')
```

Comment:
Do we also support storage resource id?

Reply:
No, because the TSI environment is accessing the storage with [data-plane API](https://docs.microsoft.com/en-us/rest/api/storageservices/) directly, instead of ARM.